### PR TITLE
p2p: implement pkg/p2p (Part I)

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
- Part I of https://github.com/pingcap/ticdc/pull/3140

### What is changed and how it works?
- Definition of new gRPC calls.
- Basic data structures for `pkg/p2p`.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/p2p/server_wrapper_test.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+	newClient = func() (p2p.CDCPeerToPeerClient, func()) {
+		conn, err := grpc.Dial(
+			addr,
+			grpc.WithInsecure(),
+			grpc.WithContextDialer(func(_ context.Context, s string) (net.Conn, error) {
+				return net.Dial("unix", addr)
+			}))
+		require.NoError(t, err)
+
+		cancel := func() {
```
[the remark this answers is not in the document]
Comment (3AceShowHand) [reply pkg/p2p/server_wrapper_test.go:92]: this one looks quite complicated, it's lucky only used in testing.

File: proto/CDCPeerToPeer.proto [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+// limitations under the License.
+
+syntax = "proto3";
+
+package p2p;
+
+option optimize_for = SPEED;
+
+service CDCPeerToPeer {
+  rpc SendMessage(stream MessagePacket) returns (stream SendMessageResponse);
```
[the remark this answers is not in the document]
Comment (liuzix) [reply proto/CDCPeerToPeer.proto:25]: These are not requests.

File: pkg/p2p/server_wrapper_test.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	serverWrapper, newClient, cancelServer := newServerWrapperForTesting(t)
+	defer cancelServer()
+
+	client, closeClient := newClient()
+	defer closeClient()
+
+	err := failpoint.Enable("github.com/pingcap/ticdc/pkg/p2p/ServerWrapperSendMessageDelay", "pause")
```
[the remark this answers is not in the document]
Comment (liuzix) [reply pkg/p2p/server_wrapper_test.go:156]: We will explore this in time. This failpoint is only used in one test case, and it is not on a hot code path, so is unlikely to be a blocking issue.

**Review by asddongmen — APPROVED** [review]

**Pull request merged**